src/script/signingprovider.cpp

```diff
@@ -73,8 +73,6 @@ void FillableSigningProvider::ImplicitlyLearnRelatedKeyScripts(const CPubKey& pu
 {
     AssertLockHeld(cs_KeyStore);
     CKeyID key_id = pubkey.GetID();
-    // We must actually know about this key already.
-    assert(HaveKey(key_id) || mapWatchKeys.count(key_id));
```

Comment:
Why is this safe to remove?

Reply:
This is only called directly after adding the privkey or watchonly key:

```
CCryptoKeyStore::AddCryptedKey
CBasicKeyStore::AddKeyPubKey
CBasicKeyStore::AddWatchOnly
```
perhaps leave a comment at the top of this function to the effect.